etl/steps/data/garden/gfs/2024-11-11/gfs_wave_one.py

```diff
@@ -0,0 +1,398 @@
+    "worry_safety": "scored_10",
+    "worthwhile": "scored_10",
+    "annual_weight1": "system",  # annual weight year 1
+    "strata": "system",
+    "psu": "system",
+    "full_partial": "binary",  # full (1) or partial (2) completed interview
+}
+
+# Define column groups
+SYSTEM_COLS = [key for key, value in COLUMNS_MAPPING.items() if value == "system"]
+DEMO_COLS = [key for key, value in COLUMNS_MAPPING.items() if value == "demographic"]
+SCORED_10_COLS = [key for key, value in COLUMNS_MAPPING.items() if value == "scored_10"]
+BINARY_COLS = [key for key, value in COLUMNS_MAPPING.items() if value == "binary"]
+CAT_COLS = [key for key, value in COLUMNS_MAPPING.items() if value.startswith("cat")]
+LOW_SCORED_COLS = [key for key, value in COLUMNS_MAPPING.items() if value in ["scored_4", "scored_5", "scored_7"]]
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow = paths.load_dataset("gfs_wave_one")
+
+    # Read table from meadow dataset.
+    tb = ds_meadow["gfs_wave_one"].reset_index()
+
+    tb["country"] = tb["country"].map(COUNTRY_MAPPING)
+
+    # cleaning nan values
+    for val in ["-98", "98", "99", -98, 98, 99, " ", ""]:
+        tb = tb.replace(val, np.nan)  # type: ignore
+
+    tb = geo.harmonize_countries(df=tb, countries_file=paths.country_mapping_path)
+
+    # Custom column: people who think their life will get better in the next 5 years
+    tb["wb_improvement"] = tb.apply(get_ineq_nan, axis=1)  # 1 if yes, 2 if no
+
+    # reverse scoring for some variables, to make them more intuitive
+    for col in ["expenses", "lonely", "worry_safety"]:
+        tb[col] = reverse_score(tb, col)
+
+    # Calculate average scores/ shares of answers for all variables
+    tb_scored_10 = average_scored(tb, cols=SCORED_10_COLS)
+    tb_share_10 = share_categorical(tb, cols=SCORED_10_COLS)
+    tb_binary = share_binary(tb, cols=BINARY_COLS + ["wb_improvement"])
+    tb_cat = share_categorical(tb, cols=CAT_COLS)
+
+    tb_scored_other = average_scored(tb, cols=LOW_SCORED_COLS)
+    tb_cat_other = share_categorical(tb, cols=LOW_SCORED_COLS)
+
+    # 97 is treated as 97 rather than 97+ (topcoded)
+    # drinks: 116 rows with maximum value of 97
+    # cigarettes: 45 rows with maximum value of 97
+    tb_scored_97 = average_scored(tb, cols=["cigarettes", "drinks"])
+
+    # Merge all tables and remove duplicate columns (na shares are calculated twice for some variables)
+    tbs_full = pr.multi_merge(
+        [tb_scored_10, tb_share_10, tb_binary, tb_cat, tb_scored_other, tb_cat_other, tb_scored_97],
+        on=["country"],
+        suffixes=["", "_DROP_COLUMN"],
+    )
+    tbs_full = tbs_full.drop(columns=[col for col in tbs_full.columns if col.endswith("_DROP_COLUMN")])
+
+    # setting the year to 2023 for all data points (>84% of data)
+    # for reference value counts: doi_annual
+    # 2023    170562
+    # 2022     32334
+    # 2024         2
+    tbs_full["year"] = 2023
```

Comment:
Perhaps an assert here just to make sure that 2023 is the biggest year group from the years